native/src/modules/layout/components/Header.js

```diff
@@ -70,35 +71,47 @@ type PropsType = {|
   goToLanguageChange?: () => void,
   routeCityModel?: CityModel,
   language: string,
+  path: string | null,
+  routeMappingType: string | null,
   dispatch: Dispatch<StoreActionType>
 |}
 
-class Header extends React.PureComponent<PropsType> {
-  canGoBackInStack (): boolean {
-    return !!this.props.previous
-  }
+const Header = (props: PropsType) => {
+  const canGoBackInStack = useCallback(() => {
+    return !!props.previous
+  })
 
-  goBackInStack = () => {
-    this.props.navigation.goBack()
-  }
+  const goBackInStack = useCallback(() => {
+    props.navigation.goBack()
+  })
 
-  goToLanding = () => {
-    const { navigation, dispatch } = this.props
+  const goToLanding = useCallback(() => {
+    const { navigation, dispatch } = props
     // $FlowFixMe Navigation type of the header does not match that of screens.
     createNavigateToLanding(dispatch, navigation)()
-  }
+  })
 
-  goToSettings = () => {
-    this.props.navigation.navigate(SETTINGS_ROUTE)
-  }
+  const goToSettings = useCallback(() => {
+    props.navigation.navigate(SETTINGS_ROUTE)
+  })
 
-  onShare = async () => {
-    const { scene, t } = this.props
-    const shareUrl = scene.route.params?.shareUrl
-    if (!shareUrl) { // The share option should only be shown if there is a shareUrl
+  const onShare = useCallback(async () => {
+    const { scene, t, path, language, routeMappingType } = props
+    const shareUrlFromScene = scene.route.params?.shareUrl
+
+    if (!shareUrlFromScene && !path && !routeMappingType) { // The share option should only be shown if there is a shareUrl
       return
     }
 
+    let shareUrl
+    if (path) {
+      shareUrl = url(path)
+    } else if (routeMappingType && routeCityModel) { // for overview pages of news, events and pois, information should be moved to routeMappings in redux store TODO in 450
+      shareUrl = cityContentUrl({ cityCode: routeCityModel.name, languageCode: language, route: routeMappingType, path: null })
+    } else { // for routes not saved in redux store eg. offers
+      shareUrl = shareUrlFromScene
+    }
```

Comment:
All of this should happen in the container. We also have there access to the navigation prop.

Reply:
moved it.